File: C/idades.c
#include <stdio.h>
#include <string.h>

void limpar_entrada() {
char c;
while ((c = getchar()) != '\n' && c != EOF) {}
}

void ler_texto(char *buffer, int length) {
fgets(buffer, length, stdin);
strtok(buffer, "\n");
}

int main() {

    char nome1[50], nome2[50];
    int idade1, idade2;
    double media;

    printf("Dados da primeira pessoa:\n");
    printf("Nome: ");
    ler_texto(nome1, 50);;
    printf("Idade: ");
    scanf("%d", &idade1);
    
    printf("Dados da segunda pessoa:\n");
    printf("Nome: ");
    limpar_entrada(); //-------- limpeza de buffer
    ler_texto(nome2, 50);
    printf("Idade: ");
    scanf("%d", &idade2);

    media = (idade1 + idade2) / 2.0;

    printf("A idade media de %s e %s eh de %.1lf anos", nome1, nome2, media);

    return 0;
}

File: Python/idades.py
print("Dados da primeira pessoa:")
nome1 = str(input("Nome: "))
idade1 = int(input("Idade: "))

print("Dados da segunda pessoa:")
nome2 = str(input("Nome: "))
idade2 = int(input("Idade: "))

media = (idade1 + idade2) / 2.0

print(f"A idade media de {nome1} e {nome2} eh de {media:.1f} anos")